src/middlewared/middlewared/plugins/crypto_/cert_profiles.py

```diff
@@ -7,6 +7,83 @@
 
 
 CERTIFICATE_PROFILES = {
+    # Options / EKUs reference rfc5246
+    'HTTPS RSA Certificate': {
+        'cert_extensions': {
+            'BasicConstraints': {
+                'enabled': True,
+                'ca': False,
+                'extension_critical': True
+            },
+            'AuthorityKeyIdentifier': {
+                'enabled': True,
+                'authority_cert_issuer': True,
+                'extension_critical': False
+            },
+            # Most TLS certs these days want "ClientAuth" these days.
+            # LetsEncrypt appears to want this extension to issue.
+            # https://community.letsencrypt.org/t/extendedkeyusage-tls-client-authentication-in-tls-server-certificates/59140/7
+            'ExtendedKeyUsage': {
+                'enabled': True,
+                'extension_critical': True,
+                'usages': [
+                    'SERVER_AUTH',
+                    'CLIENT_AUTH',
+                ]
+            },
+            # RSA certs need "digitalSignature" for DHE,
+            # and "keyEncipherment" for nonDHE
+            # Include "keyAgreement" for compatibility (DH_DSS / DH_RSA)
+            # See rfc5246
+            'KeyUsage': {
+                'enabled': True,
+                'extension_critical': True,
+                'digital_signature': True,
+                'key_encipherment': True,
+                'key_agreement': True,
+            }
+        },
+        'key_length': 2048,
+        'key_type': 'RSA',
+        'lifetime': NOT_VALID_AFTER_DEFAULT,
```

Comment:
I assume you want to use `DEFAULT_LIFETIME_DAYS` instead of `NOT_VALID_AFTER_DEFAULT` ?

Reply:
That was copied from the OpenVPN certificate profiles, but I can make that change.

Reply:
`NOT_VALID_AFTER_DEFAULT ` has not been initialized and does not exist which will result in this part erroring out as the variable does not exist